test/integration/framework/utils.go

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	kubecfg    = "kubecfg"
+	kubeconfig = "kubeconfig"
+	password   = "password"
+)
+
+// InitializeTestSeedClients will use the Garden Kubernetes client to read the Seed Secret in the Garden
+// cluster which contains a Kubeconfig that can be used to authenticate against the Seed cluster. With it,
+// a Kubernetes client for the Seed cluster will be initialized and attached to
+// the already existing GardenTestOperation object.
+func (o *GardenerTestOperation) initializeTestSeedClients() error {
```

Comment:
why not just `initializeSeedClients` - I mean we know that this is a test, we don't have to put it everywhere.

Reply:
because its different

Reply:
Yes, but this thing isn't inheriting from `Operation`, is it? Then, you should still just call it `initializeSeedClients`.